presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -131,13 +136,17 @@ public void setUp()
                 daemonThreadsNamed("test-executor-%s"),
                 new ThreadPoolExecutor.DiscardPolicy());
         scheduledExecutor = newScheduledThreadPool(2, daemonThreadsNamed("test-scheduledExecutor-%s"));
+        partitioningProviderManager = new PartitioningProviderManager();
+        session = testSessionBuilder().build();
     }
 
     @AfterMethod(alwaysRun = true)
     public void tearDown()
     {
         executor.shutdownNow();
         scheduledExecutor.shutdownNow();
+        partitioningProviderManager = null;
```

Comment:
I see this pattern in a few other places. Curious why this is required at `tearDown`?

Reply:
TestNG instantiates all test classes eagerly at the very beginning. This prevents us from running out of memory (creating object needed for a test right before the test, and nullifying them right afterwards)